Networking/WebSockets/WebSocketImpl.cc

```diff
@@ -362,6 +369,14 @@ namespace litecore { namespace websocket {
 
     // Initiates a request to close the connection cleanly.
     void WebSocketImpl::close(int status, fleece::slice message) {
+        if(!_didConnect) {
+            // The web socket is being requested to close before it's even connected, so just
```

Comment:
This should just call `closeSocket()` so that the underlying socket disconnects. Then it doesn't have to do the stuff below, since `onClose` will be called.

Reply:
Also discussed this offline, but in the end yes I agree with this!  Had to modify the test for this but it's a sane adjustment.